### PR TITLE
ENH: Add single point versions of transform, fwd, inv

### PR DESCRIPTION
**Pull request by greglucas**
When transforming single points, this can add a significant speed benefit.

 - [x] Tests added
 - Fully documented
 - [ ] including `history.rst` for all changes
 - [x] `api/*.rst` for new API

### CODE REVIEW

File: pyproj/geod.py [hunk 1]
```diff
@@ -310,6 +311,61 @@ def fwd(  # pylint: disable=invalid-name
...
+        Accepted numeric scalar:
+
+        - :class:`int`
+        - :class:`float`
+        - :class:`numpy.floating`
+        - :class:`numpy.integer`
+
+        Parameters
+        ----------
+        lon: scalar or array
```
Comment (snowman2) [inline pyproj/geod.py:338]: I think this is something to look out for in the new docstrings.
```suggestion
        lon: scalar
```

File: test/test_transformer.py [hunk 2]
```diff
@@ -1505,6 +1505,18 @@ def test_4d_transform__inplace__numpy__int():
     assert tarr[0] == 2019
 
 
+def test_transform_single_point():
```
Comment (snowman2) [inline test/test_transformer.py:1508]: Might not be a bad idea to parametrize `z` and `t` in this test for the cases that they are None (i.e. not provided).

File: pyproj/transformer.py [hunk 3]
```diff
@@ -820,6 +821,108 @@ def transform(  # pylint: disable=invalid-name
...
+        radians: bool = False,
+        errcheck: bool = False,
+        direction: Union[TransformDirection, str] = TransformDirection.FORWARD,
+    ) -> Tuple[float, float]:
+        ...
+
+    @overload
+    def transform_point(  # pylint: disable=invalid-name
+        self,
+        x: float,
```
Comment (snowman2) [inline pyproj/transformer.py:838]: It would be good for theses kwargs to be consistent with `transform`.

**Comment by snowman2** [comment]
See #1204 for cython version of `Transformer.transform_point`.

**Comment by greglucas** [comment]
Closing in favor of #1204. Will open a separate PR for the Geod point functions.

**Comment by idanmiara** [comment]
> Closing in favor of #1204. Will open a separate PR for the Geod point functions.

https://github.com/pyproj4/pyproj/pull/1206

Just adding the reference here. Thanks for this work @greglucas @snowman2 !

**Pull request closed without merging**